redaxo/src/addons/media_manager/lib/effects/effect_convert2img.php

```diff
@@ -72,9 +72,19 @@ public function execute()
 
         $to_path = rex_path::addonCache('media_manager', 'media_manager__convert2img_' . md5($this->media->getMediaPath()) . '_' . $filename_wo_ext . $convert_to['ext']);
 
-        $cmd = $convert_path . ' -density '.$density.' "' . $from_path . '[0]" -colorspace RGB "' . $to_path . '"';
-
-        exec($cmd, $out, $ret);
+		if ($convert_path == 'imagick') {
```

Comment:
von der logik hier ist es jetzt so programmiert dass imagick präferiert wird, wenn beides vorhanden ist..

liefert Imagick die besseren ergebnisse, oder sollte es eher der fallback sein?

Reply:
Ok, kann man bei Bedarf. tauschen
Allerdings habe ich ich die Erfahrung gemacht, dass exec bei vielen Hostern nicht verfügbar ist und meist in PHP7 das imagick mit installiert wird.
Qualitativ kann ich nichts dazu sagen.

Reply:
Ich denke, die Ergebnisse sollten gleich sein, denn `convert` ist ja imagemagick, und die php-extension sollte letztlich die gleiche lib verwenden.

Ich würde die PHP-Variante bevorzugt verwenden. Denn neben fehlenden exec-Rechten gibt es auch noch die Variante, dass der `convert`-Pfad gar nicht erst gefunden wird. 
So ist lokal bei mir mit meinem homebrew-setup, da kennt php aktuell nicht meine vollständige PATH-Variable, und findet es deswegen nicht. (ließe sich bestimmt auch irgendwie lösen).

ich würde also als pseudo-code grob das machen:

```
if (class_exists(Imagick)) {
   new Imagick
   ...
} else {
    getConvertPath
    exec
}